assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -95,6 +96,28 @@ public void newAssignment(Assignment assignment) {
         }
     }
 
+    @Override
+    @Transactional
+    public void newAssignmentTimeSheet(AssignmentTimeSheet timeSheet) {
+        if (!existsAssignmentTimeSheet(timeSheet.getId())) {
+            sessionFactory.getCurrentSession().persist(timeSheet);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void deleteAssignmentTimeSheet(AssignmentTimeSheet timeSheet) {
+        if(timeSheet!= null) {
+        	sessionFactory.getCurrentSession().delete(sessionFactory.getCurrentSession().merge(timeSheet));
+        }
+    }
+
+    @Override
+    @Transactional
+    public boolean existsAssignmentTimeSheet(Long timeSheetId) {
```

Comment:
this method should likely take the TimeSheet object not an id

Reply:
`public boolean existsAssignmentTimeSheet(AssignmentTimeSheet timeSheet) {`
done.